packages/slice-machine/lib/builders/SliceBuilder/index.tsx

```diff
@@ -109,41 +96,42 @@ const SliceBuilder: React.FC<SliceBuilderProps> = ({
 
   if (!variation || !sliceView) return null;
 
-  const onTakingCustomScreenshot = () => {
-    // eslint-disable-next-line @typescript-eslint/no-misused-promises
-    checkSimulatorSetup(true, async () => {
-      await generateScreenShot(
+  const onTakingSliceScreenshot = () => {
+    generateSliceScreenshot(variation.id, extendedComponent.component, setData);
+  };
+
+  const onTakingSliceCustomScreenshot = (file: Blob) => {
+    checkSimulatorSetup(true, () =>
```

Comment:
seems to be a small mistake, the simulator needs to be verified for SliceScreenshot, not CustomScreenshot.

Custom Screenshot are a custom file directly uploaded and used.
While Slice Screenshots are taken using the user's project so we need to verify it has been setup properly.

Reply:
True, perhaps we could investigate this in a spike too see why it was originally there and if adding it would lead to any side effects